samples/bluetooth/hci_uart_3wire/src/main.c

```diff
@@ -224,6 +225,12 @@ static void process_unack(void)
 		net_buf_unref(buf);
 		unack_queue_len--;
 		number_removed--;
+
+		acked = true;
+	}
+
+	if (acked) {
+		k_poll_signal_raise(&tx_queue_change, 0);
```

Comment:
Just a suggestion to avoid repeated assignment.

```c
if (!number_removed) {
  return;
}

do {
  ...
} while (number_removed);

k_poll_signal_raise(&tx_queue_change, 0);
```